### DISCUSSION TITLE
uprobes with golang binaries not working (ERROR: Could not resolve symbol)

### DISCUSSION THREAD

**Discussion by wunderwuzzi23** [bpftrace/bpftrace#2017] [Q&A] [answered]
Hello everyone, I'm pretty new to `bpftrace` and been trying to trace Golang binaries - so far without any luck. There is a symbols error that `bpftrace` displays: `ERROR: Could not resolve symbol`. And I'm not sure what the issue is exactly.

Here are the repro steps:

1) Golang program `main.go`

```
package main

import (
	"fmt"
)

//go:noinline
func initialize(s string) {
	fmt.Println("Init called " + s)
}

func main() {
	initialize("test")
}
```

2) Then I compiled it using `go build`.

3) Looking at the symbols using `objdump -t GoTest | grep main.init` I get:
```
objdump -t GoTest | grep initialize
000000000047e1e0 g     F .text  0000000000000096 main.initialize
```

4) Then trying to run `bpftrace` using
```
sudo bpftrace -e 'uprobe:GoTest:main.initialize { printf("hi"); }'
Attaching 1 probe...
ERROR: Could not resolve symbol: GoTest:main.initialize
```

Any idea what I'm doing wrong? Any input is much appreciated.

Go version is:

```
go version go1.17.1 linux/amd64
```

**Comment by fbs** [comment]
Can you list the symbols using bpftrace? `bpftrace -l 'uprobe:GoTest'`

**Comment by wunderwuzzi23** [comment]
Thanks for offering to help, `bpftrace -l` shows it:

```
$ sudo bpftrace -l 'uprobe:GoTest:*' | grep initialize
uprobe:GoTest:main.initialize
```

Also, in case it matters version is `bpftrace v0.13.0`.

**Comment by fbs** [comment, marked as answer]
Seems like you need to use 'uprobe:./GoTest:main.initialize'

```
vagrant@ubuntu-hirsute:~/build$ cat main.go
package main

import (
	"fmt"
)

//go:noinline
func initialize(s string) {
	fmt.Println("Init called " + s)
}

func main() {
	initialize("test")
}
vagrant@ubuntu-hirsute:~/build$ go build main.go
vagrant@ubuntu-hirsute:~/build$ ./main
Init called test
vagrant@ubuntu-hirsute:~/build$ sudo bpftrace -l 'uprobe:./main:main.init*'
uprobe:./main:main.initialize
vagrant@ubuntu-hirsute:~/build$ sudo bpftrace -e 'uprobe:./main:main.initialize { print("im hit\n"); }' -c ./main
Attaching 1 probe...
Init called test
im hit
```

**Comment by wunderwuzzi23** [comment]
Thanks so much! That works!
